pkg/controller/controllerutil/controllerutil.go

```diff
@@ -309,6 +309,20 @@ func CreateOrPatch(ctx context.Context, c client.Client, obj client.Object, f Mu
 	if (hasBeforeStatus || hasAfterStatus) && !reflect.DeepEqual(beforeStatus, afterStatus) {
 		// Only issue a Status Patch if the resource has a status and the beforeStatus
 		// and afterStatus copies differ
+		if result == OperationResultUpdated {
+			// If Status was replaced by Patch before, set it to afterStatus
+			objectAfterPatch, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&obj)
```

Comment:
Should this be `ToUnstructured(obj)` (no address operator) like L282?

I see that L257 makes a deep copy while L282 does not. Is it possible that `Patch()` on L303 has manipulated `afterStatus` as well? I think a test that calls `CreateOrPatch()` with an `*unstructured.Unstructured` will confirm.

Reply:
Probably should use `obj`, not its address, thanks @cbandy, fixed. But tests both variants, it seems to be working fine.